indico/modules/rb_new/controllers/backend/blockings.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from collections import defaultdict
+
+from flask import jsonify, request, session
+from webargs import fields
+from webargs.flaskparser import use_args, use_kwargs
+from werkzeug.exceptions import Forbidden
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.util.session import no_autoflush
+from indico.modules.groups import GroupProxy
+from indico.modules.rb.controllers import RHRoomBookingBase
+from indico.modules.rb.models.blocked_rooms import BlockedRoom
+from indico.modules.rb.models.blockings import Blocking
+from indico.modules.rb.models.rooms import Room
+from indico.modules.rb.notifications.blockings import notify_request
+from indico.modules.rb_new.operations.blockings import get_room_blockings
+from indico.modules.rb_new.schemas import blockings_schema
+from indico.modules.users.models.users import User
+from indico.web.util import jsonify_data
+
+
+def _group_id_or_name(principal):
+    provider = principal['provider']
+    if provider is None or provider == 'indico':
+        return principal['id']
+    else:
+        return principal['name']
+
+
+@no_autoflush
+def _populate_blocking(blocking, room_ids, allowed_principals, reason):
+    blocking.blocked_rooms = [BlockedRoom(room_id=room.id) for room in Room.query.filter(Room.id.in_(room_ids))]
+    blocking.reason = reason
+    blocking.allowed = [GroupProxy(_group_id_or_name(pr), provider=pr['provider'])
+                        if pr.get('is_group')
+                        else User.get_one(pr['id'])
+                        for pr in allowed_principals]
+
+
+def create_blocking(room_ids, allowed_principals, start_date, end_date, reason, created_by):
```

Comment:
aren't those *operations*?

Reply:
Yeah, I thought I'd keep them here since they're not used anywhere else and wanted to keep the operations mode minimal. But maybe it makes more sense to have them all there.